cupyx/scipy/fft/fft.py

```diff
@@ -0,0 +1,419 @@
+import cupy
+from cupy.cuda import cufft
+from cupy.fft.fft import (_fft, _default_fft_func, hfft as _hfft,
+                          ihfft as _ihfft)
+from cupy.fft.fft import fftshift, ifftshift, fftfreq, rfftfreq
+import numpy as np
+from numbers import Number
+
+
+__all__ = ['fft', 'ifft', 'fft2', 'ifft2', 'fftn', 'ifftn',
+           'rfft', 'irfft', 'rfft2', 'irfft2', 'rfftn', 'irfftn',
+           'hfft', 'ihfft',
+           'fftshift', 'ifftshift', 'fftfreq', 'rfftfreq']
+
+try:
+    import scipy.fft as _scipy_fft
+except ImportError:
+    class _DummyModule:
+        def __getattr__(self, name):
+            return None
+
+    _scipy_fft = _DummyModule()
+
+# Backend support for scipy.fft
+
+__ua_domain__ = 'numpy.scipy.fft'
+_implemented = {}
+
+
+def __ua_convert__(dispatchables, coerce):
+    if not all(d.type == np.ndarray for d in dispatchables):
+        return NotImplemented
+
+    if coerce:
+        try:
+            replaced = [cupy.asarray(d.value) if d.coercible else d.value
+                        for d in dispatchables]
+        except TypeError:
+            return NotImplemented
+    else:
+        replaced = [d.value for d in dispatchables]
+
+    if not all(isinstance(r, cupy.ndarray) for r in replaced):
+        return NotImplemented
+
+    return replaced
+
+
+def __ua_function__(method, args, kwargs):
+    fn = _implemented.get(method, None)
+    if fn is None:
+        return NotImplemented
+    return fn(*args, **kwargs)
```

Comment:
Try/except here, return `NotImplemented` on `TypeError`.

Reply:
This still holds, with the newly-vendored `uarray` that I merged last night.

Reply:
That would also silence genuine errors though e.g. `fft(x, axis='')`

Reply:
It would be better to replace it with raising a `BackendNotImplementedError`, then.

Reply:
I still can't see the reason for doing this, all I can see is that it will silence (or at least disguise) genuine errors. In fact I notice that you don't do it yourself in `unumpy`'s [`cupy` backend]. 

Could you give an example where `TypeError` would be raised that isn't just a bug in the caller?

> It would be better to replace it with raising a `BackendNotImplementedError`, then.

We can't raise `BackendNotImplementedError` without adding `uarray` as a dependency.

[`cupy` backend]: https://github.com/Quansight-Labs/unumpy/blob/master/unumpy/cupy_backend.py